src/JustSaying/JustSayingBus.cs

```diff
@@ -296,9 +342,102 @@ public InterrogationResult Interrogate()
         });
     }
 
+    /// <inheritdoc/>
     public void Dispose()
     {
         _startLock?.Dispose();
         _loggerFactory?.Dispose();
     }
+
+    /// <inheritdoc/>
+    public Task PublishAsync(IEnumerable<Message> messages, PublishBatchMetadata metadata, CancellationToken cancellationToken)
```

Comment:
I wonder if we should make this generic to limit messaage batch publishing to one type at a time?

To me this gives the appearance of atomicity but we don't actually guarantee that. 
(Ie if I PublishAsync([MessageType1, MessageType2]...) our interface gives me the false sense that if publishing for MessageType1 fails, publishing for MessageType2 will also fail.)

Reply:
If we do want to keep the multiple message type publishing, I think we should cover that use-case in tests.

Reply:
I've added tests for this. I kept the method non-generic (`IEnumerable<Message>`) to match the non-batch publishing method so there was symmetry of usage. Otherwise if someone had this:

```csharp
foreach (var message in messages)
{
    await publisher.PublishAsync(message);
}
```

they couldn't just trivially change to:

```csharp
await publisher.PublishAsync(messages);
```

In that scenario someone would need to handle partial publishing anyway. Otherwise the caller needs to publish the specific concrete type and do their own grouping themselves, which is _probably_ going to need further refactoring in their app to use batching compared to a simple swap-over.